File: LeetCode/Problems/Python/202. Happy Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        
        def getNext(n):
            temp = 0
            while n > 0:
                digit = n % 10
                temp += digit**2
                n //= 10
            return temp
        
        lookup = set()
        
        while n!=1 and n not in lookup:
            lookup.add(n)
            n = getNext(n)
           
        return n==1

File: LeetCode/Problems/C++/202. Happy Number.cpp
class Solution {
public:
    int getNext(int n){
        int total_sum = 0, digit;
        while(n>0){
            digit = n % 10;
            total_sum += digit * digit;
            n /=10;
        }
        return total_sum;
    }
    bool isHappy(int n) {
        unordered_set<int> seen;
        
        while(n!=1 && seen.find(n)==seen.end()){
            seen.insert(n);
            n = getNext(n);
        }
        return n==1;
    }
};